### PR TITLE
Fixes #154: test case added for generate script

### LINKED ISSUE #154: Add integration test for documentation generation
### This is a
- [ ] Bug report
- [x] Feature request

### Expected behaviour

Add unit test for documentation generation. it should contain following test case.
- generate script should return exit code 0
- documentation zip file has to be created in the temp folder
- preview folder has to be created in temp folder

### Steps to reproduce
1.
2.
3.

### PR DESCRIPTION
**Pull request by sch00lb0y**
## Description

test case added generate script. following test cases added
- generate script should return exit code 0
- documentation zip file has to be created in the temp folder
- preview folder has to be created in temp folder

## Related Issue

close #154
## Screenshots (if appropriate):
generator test
## Types of changes

- [ ] Bug fix 
- [x] New feature

## Checklist:

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only one commit per issue.

### CODE REVIEW

File: tests/test.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+const spawn = require('child_process').spawn
+const uuidV4 = require("uuid/v4")
+const fs = require('fs')
+
+describe('WebUi generator', () => {
+    
+    let uniqueId = uuidV4()
+    let email = "rbalajis25@gmail.com"
+    let args = [
+        "-g", "https://github.com/sch00lb0y/tempCheck.git",
```
Comment (imujjwal96) [inline tests/test.js]: This test is dependent on your email and your repository. The day you delete your repository, this test will fail.
Please write tests that are not dependent of external factors.

Comment (sch00lb0y) [inline tests/test.js]: I have made changes

File: tests/test.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+const assert = require('assert')
+const spawn = require('child_process').spawn
+const uuidV4 = require("uuid/v4")
+const fs = require('fs')
+
+describe('WebUi Generator', () => {
+  let uniqueId = uuidV4()
+  let email = 'fossasia@gmail.com'
+  let args = [
+    "-g", "https://github.com/fossasia/yaydoc.git",
```
Comment (imujjwal96) [inline tests/test.js]: just to be sure, does running this test actually executes the `generate` script?

Comment (sch00lb0y) [inline tests/test.js]: I'm spanning the process in before function

Comment (imujjwal96) [inline tests/test.js]: it does or it doesn't?

Comment (sch00lb0y) [inline tests/test.js]: yeah, it does. here is the output
```
  WebUi Generator
    ✓ exit code shoul be zero
    ✓ preview folder should be created
    ✓ zipped documentation should be created
  3 passing (2m)
``` 
i didn't add the `npm test` for this pr. once my previous pr got merged. This is good to go with travis

Comment (imujjwal96) [inline tests/test.js]: OK. Are you performing cleanup of these generated docs?

Comment (sch00lb0y) [inline tests/test.js]: No, since the test is running on Travis. I don't think so we need cleaning.

File: tests/test.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  let files
+  before((done) => {
+    let process = spawn('./generate.sh', args)
+    process.on('exit', (code) => {
+      exitCode = code
+      files = fs.readdirSync(`temp/${email}`)
+      done()
+    })
+  })
+  it('exit code shoul be zero', () => {
```
Comment (pri22296) [inline tests/test.js]: fix typo here

File: .travis.yml [hunk 4]
```diff
@@ -13,6 +13,7 @@ env:
 
 install:
   - npm install
+  - sudo pip install -r requirements.txt
```
Comment (pri22296) [inline .travis.yml]: I would suggest you to instead set the language in travis as python and install node via `nvm install node`. That way both python and node would be able to run on travis properly.

Comment (sch00lb0y) [inline .travis.yml]: I have tried but it is not working but when I'm having nodejs as language and `sudo pip install` the build is passing

Comment (pri22296) [inline .travis.yml]: weird. it seems to work at https://github.com/GluuFederation/gluu-webui/blob/master/.travis.yml . Anyways use whatever works for now

**Comment by sch00lb0y** [comment]
@pri22296 I have tested and works fine on travis. here is the [link](https://travis-ci.org/fossasia/yaydoc/builds/245248436) of Travis build. please check

**Comment by pri22296** [comment]
@sch00lb0y LGTM. Please answer @imujjwal96 's query!

**Comment by sch00lb0y** [comment]
@imujjwal96 can you elaborate your query?

File: .travis.yml (hunk 4 above)
Comment (imujjwal96) [inline .travis.yml]: Why do we pip install here?

Comment (pri22296) [inline .travis.yml]: That would eventually be needed for #151

**Pull request merged**